src/Level06/AppM.hs

```diff
@@ -6,90 +6,78 @@ module Level06.AppM where
 import           Control.Monad.Except   (MonadError (..))
 import           Control.Monad.IO.Class (MonadIO (..))
 
-import           Data.Text              (Text)
-
 import           Level06.Types          (Error)
 
-import           Data.Bifunctor         (first)
+-- | We're going to upgrade the capability of our AppM by generalising the type of the errors that it
+-- handles. This means that we'll be able to reuse our 'AppM' in more places that maybe have an
+-- overabundance of 'IO (Either e a)' types.
 
--- We're going to add a very useful abstraction to our application. We'll
--- automate away the explicit error handling and inspection of our Either values
--- while preserving the type-level information that tells us what can go wrong.
---
--- To do this we will create a newtype `AppM` that is a shorthand way of
--- describing the return type of a function that may contain an error.
---
--- This will work in the same manner as the Functor/Applicative/Monad
--- instances for Either, with functions being applied to the Right value and
--- everything been ignored if a Left value is encountered, returning that Left
--- value.
+-- | The first step is to add the type parameter to our 'AppM', which we will also rename to
+-- 'AppM'', pronounced "AppM prime". We do this because we don't want to have to refactor our entire
+-- application to simply repeat our 'Error' type everywhere. We have a far simpler solution.
 --
--- f <$> (Left e)  = Left e
--- f <$> (Right a) = Right (f a)
+-- Our new 'AppM'' will also use the record syntax to define our 'runAppM' function. This is a more
+-- common definition of this kind of newtype.
 --
--- (Left e)  >>= f = Left e
--- (Right a) >>= f = f a
---
--- This means when we have a function doing this sort of shuffling:
+newtype AppM' e a = AppM'
+  { runAppM' :: IO (Either e a)
+  }
+
+-- | Predominantly our application has only one error type: 'Error'. It would be tedious to have to
+-- declare that on every signature. We're able to use a type _alias_ to avoid this problem. We can
+-- define this type alias to make the error type variable concrete as 'Error'.
 --
--- foo :: IO (Either Error Value)
--- foo = do
---   aE <- mightFail
---   either (pure . Left) needsAButMightFail aE
---   where
---     mightFail :: IO (Either Error Int)
---     alsoMightFail :: Int -> IO (Either Error Value)
+type AppM = AppM' Error
```

Comment:
Should `AppM'` be `AppM` and `AppM` be `App`?

Reply:
That's a nicer naming scheme, I've updated Level06/07 to match.